app/cdap/components/WranglerGrid/SelectColumnPanel/utils.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {
+  IHeaderNamesList,
+  IMultipleSelectedFunctionDetail,
+} from 'components/WranglerGrid/SelectColumnPanel/types';
+import { MULTI_SELECTION_COLUMN } from 'components/WranglerGrid/SelectColumnPanel/constants';
+
+export const getColumnsSupportedType = (
+  transformationDataType: string[],
+  columnsList: IHeaderNamesList[]
+) => {
+  return transformationDataType?.length > 0 && transformationDataType?.includes('all')
+    ? transformationDataType?.filter((supportedType: string) => supportedType === 'all')
+    : columnsList?.filter((columnDetail: IHeaderNamesList) => {
+        return transformationDataType?.some((dataTypeCollection: string | string[]) => {
+          return dataTypeCollection?.includes(columnDetail?.type[0]?.toLowerCase());
+        });
+      });
+};
+
+export const getFilteredColumn = (
+  transformationDataType: string[],
+  columnsList: IHeaderNamesList[]
+) => {
+  return transformationDataType?.length > 0 && transformationDataType?.includes('all')
+    ? columnsList
+    : columnsList?.filter((columnDetail: IHeaderNamesList) => {
+        return transformationDataType?.some((dataTypeCollection: string | string[]) => {
+          return dataTypeCollection?.includes(columnDetail?.type[0]?.toLowerCase());
+        });
+      });
+};
+
+export const enableDoneButton = (transformationName, selectedColumns) => {
+  if (
+    MULTI_SELECTION_COLUMN.filter(
+      (functionNameDetail: IMultipleSelectedFunctionDetail) =>
+        functionNameDetail.value === transformationName && !functionNameDetail.isMoreThanTwo
+    )?.length
+  ) {
+    return selectedColumns?.length === 2 ? false : true;
+  } else if (
+    MULTI_SELECTION_COLUMN.filter(
```

Comment:
??? isn't this the same as the one above?

Reply:
All the three conditions are mandatory
1. first If implies that the button should be enabled on when two columns are selected
2. second else If implies that even if one column is selected the button should be enabled (more than two column selection)
3. third else implies case of single selection of radio button only one needs to be selected to enable button
Added above comments in the file also